Library/Homebrew/tab.rb

```diff
@@ -30,36 +30,51 @@ class Tab
 
   attr_accessor :homebrew_version, :tabfile, :built_as_bottle,
                 :changed_files, :loaded_from_api, :time, :stdlib, :aliases, :arch, :source,
-                :built_on
+                :built_on, :caskfile_only, :uninstall_artifacts
   attr_writer :used_options, :unused_options, :compiler, :source_modified_time
 
-  # Returns the formula's runtime dependencies.
+  # Returns the formula or cask runtime dependencies.
   #
   # @api internal
   attr_writer :runtime_dependencies
 
   # Instantiates a {Tab} for a new installation of a formula.
   def self.create(formula, compiler, stdlib)
-    build = formula.build
-    runtime_deps = formula.runtime_dependencies(undeclared: false)
-    attributes = {
+    # TODO: uncomment for Homebrew 4.4.0
+    # odeprecated "Tab.create", "Tab.create_from_formula"
+
+    create_from_formula(formula, compiler, stdlib)
+  end
+
+  def self.generic_attributes(formula_or_cask)
+    {
       "homebrew_version"        => HOMEBREW_VERSION,
-      "used_options"            => build.used_options.as_flags,
-      "unused_options"          => build.unused_options.as_flags,
-      "tabfile"                 => formula.prefix/FILENAME,
-      "built_as_bottle"         => build.bottle?,
       "installed_as_dependency" => false,
       "installed_on_request"    => false,
-      "poured_from_bottle"      => false,
-      "loaded_from_api"         => false,
+      "loaded_from_api"         => formula_or_cask.loaded_from_api?,
       "time"                    => Time.now.to_i,
-      "source_modified_time"    => formula.source_modified_time.to_i,
-      "compiler"                => compiler,
-      "stdlib"                  => stdlib,
-      "aliases"                 => formula.aliases,
-      "runtime_dependencies"    => Tab.runtime_deps_hash(formula, runtime_deps),
+      "source_modified_time"    => 0,
```

Comment:
Just flagging: this doesn't currently work for casks since we don't have `source_modified_time` implemented there. I tried to get it working, but it wasn't as straight forward as I hoped. I plan on opening a follow-up PR after to implement this.

Reply:
I'd suggest the field is omitted rather than `0` until it works.